app/models/miq_cockpit_ws_worker/authenticator.rb

```diff
@@ -0,0 +1,53 @@
+class MiqCockpitWsWorker::Authenticator
```

Comment:
Were you going to move this to a module that gets included?  I ask because classes are usually for encapsulating state and behavior, this is strictly behavior.  I would make this a module and either include it in the class utilizing it or use composition to initialize the MiqCockpitWsWorker with this as the authenticator and use delegation to delegate these methods in MiqCockpitWsWorker to this "authenticator".